generators/add-action/asset-compute/index.js

```diff
@@ -0,0 +1,94 @@
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+// #########################################################################################
+// 3rd party template
+// #########################################################################################
+
+const path = require('path')
+const ActionGenerator = require('../../../lib/ActionGenerator')
+
+class AssetComputeGenerator extends ActionGenerator {
+  constructor (args, opts) {
+    super(args, opts)
+    this.props = {}
+  }
+
+  async prompting () {
+    this.props.actionName = await this.promptForActionName('example of a custom worker for the Adobe Asset Compute service', 'worker-example')
+  }
+
+  writing () {
+    this.sourceRoot(path.join(__dirname, './templates'))
+
+    this.addAction(this.props.actionName, './_worker.js', {
+      tplContext: this.props,
+      dependencies: {
+        '@adobe/asset-compute-sdk': '^1.0.2'
+      },
+      devDependencies: {
+        '@adobe/aio-cli-plugin-asset-compute': '^1.3.0'
+      },
+      dotenvStub: {
+        label: 'please provide the following environment variables for the Asset Compute devtool. You can use AWS or Azure, not both:',
+        vars: [
+          'ASSET_COMPUTE_INTEGRATION_FILE_PATH',
+          'S3_BUCKET',
+          'AWS_ACCESS_KEY_ID',
+          'AWS_SECRET_ACCESS_KEY',
+          'AWS_REGION',
+          'AZURE_STORAGE_ACCOUNT',
+          'AZURE_STORAGE_KEY',
+          'AZURE_STORAGE_CONTAINER_NAME'
+        ]
+      },
+      actionManifestConfig: {
+        inputs: { LOG_LEVEL: 'debug' },
+        annotations: { 'require-adobe-auth': true }
+      }
+    })
+
+    // modify the package.json to contain the Asset Compute testing and development tools
+    const packagejsonPath = this.destinationPath('package.json')
+    const packagejsonContent = this.fs.readJSON(packagejsonPath)
+
+    // add asset compute worker-tests in packages.json scripts
+    if (!packagejsonContent.scripts) {
+      packagejsonContent.scripts = {}
+      packagejsonContent.scripts.test = 'adobe-asset-compute test-worker'
+      packagejsonContent.scripts.debug = 'app run && adobe-asset-compute devtool'
+    } else {
+      if (packagejsonContent.scripts.test && !packagejsonContent.scripts.test.includes('adobe-asset-compute test-worker')) {
+        packagejsonContent.scripts.test = packagejsonContent.scripts.test.concat(' && adobe-asset-compute test-worker')
+      } else {
+        packagejsonContent.scripts.test = 'adobe-asset-compute test-worker'
+      }
+
+      if (packagejsonContent.scripts.debug && !packagejsonContent.scripts.debug.includes('aio app run && adobe-asset-compute devtool')) {
+        packagejsonContent.scripts.debug = packagejsonContent.scripts.debug.concat(' && aio app run && adobe-asset-compute devtool')
+      } else {
+        packagejsonContent.scripts.debug = 'aio app run && adobe-asset-compute devtool'
```

Comment:
just so I have this clear for the documentation, to run the devtool, we will run `npm run debug`?

Reply:
also, `aio app run` will open up a browser and start a local server and `adobe-asset-compute devtool` will also open up a browser and run a local server...are we sure this works with a simple `&&`? I cannot seem to get this work.

 We may need to instead do this: `aio app deploy && adobe-asset-compute devtool` or run a concurrent process like we do in the devtool: https://github.com/adobe/asset-compute-devtool/blob/master/server/package.json#L17

cc/ @alexkli

Reply:
@jdelbick yeah, no, this must use an oclif/aio app hook instead. From [here](https://jira.corp.adobe.com/browse/NUI-561?focusedCommentId=21602348&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-21602348):

```
"scripts": {
    "pre-app-run": "echo pre-app-run",
    "post-app-run": "echo post-app-run",
    "pre-app-build": "echo pre-app-build",
    "post-app-build": "echo post-app-build",
    "pre-app-deploy": "echo pre-app-deploy",
    "post-app-deploy": "echo post-app-deploy"
  }
```

So we need to use `pre-app-run` or `post-app-run`. The challenge of two blocking commands with open servers still remains, but maybe there is a solution? @purplecabbage any ideas?

Reply:
I just tried out:
```
    "post-app-run": "adobe-asset-compute devtool"
```
and it works! See [here](https://oclif.io/docs/hooks) oclif runs hooks in parallel: "Multiple hooks are run in parallel. This behavior may change in a future release."

Reply:
Is there a roadmap for oclif? If there is one, look if there is a planned behavior change for this. Or do they say more in any of their github issues/forums/... ?

`spawn` (using processes) may be another way to go if we want to do the parallel run manually. But I would urge to refrain from it as long as we don't need to, in order to avoid "usual" parallelism pitfalls. 

Also... v14 of node introduces worker threads: https://nodejs.org/dist/latest-v14.x/docs/api/worker_threads.html. This would be the perfect candidate to solve the parallelism issue if it really arises.

Reply:
_If_ oclif would change... but until they do, we can stick to this solution. (I assume oclif would be forced to come up with a backwards compatible solution anyway).

Reply:
Hence my roadmap question :). I think that is a change they would announce. So we're good for now.
(Don't over-engineer).